Error installing

Run through the steps until this step:
```bash
$ make install
git submodule -q foreach --recursive git reset -q --hard
git submodule update --init --recursive
if patch --dry-run -N -d /home/ant/go/src/gopkg.in/libgit2/git2go.v27 -p1 < patch/git2go.v27.patch >/dev/null; then \
        patch -d /home/ant/go/src/gopkg.in/libgit2/git2go.v27 -p1 < patch/git2go.v27.patch; \
fi
patching file script/build-libgit2-static.sh
make -C /home/ant/go/src/gopkg.in/libgit2/git2go.v27 install-static
make[1]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27'
./script/build-libgit2-static.sh
+ VENDORED_PATH=vendor/libgit2
+ cd vendor/libgit2
+ mkdir -p install/lib
+ mkdir -p build
+ cd build
+ cmake -DTHREADSAFE=ON -DBUILD_CLAR=OFF -DBUILD_SHARED_LIBS=OFF -DCMAKE_C_FLAGS=-fPIC -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=../install -DUSE_EXT_HTTP_PARSER=OFF -DUSE_HTTPS=OFF -DUSE_NSEC=OFF -DUSE_SSH=OFF -DCURL=OFF -DUSE_GSSAPI=OFF -DUSE_BUNDLED_ZLIB=ON ..
./script/build-libgit2-static.sh: 11: ./script/build-libgit2-static.sh: cmake: not found
Makefile:11: recipe for target 'build-libgit2' failed
make[1]: *** [build-libgit2] Error 127
make[1]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27'
Makefile:34: recipe for target 'build-libgit2' failed
make: *** [build-libgit2] Error 2
```

Thank you @isacikgoz. That got me further but am not hit with: 
```bash
 make install
git submodule -q foreach --recursive git reset -q --hard
git submodule update --init --recursive
if patch --dry-run -N -d /home/ant/go/src/gopkg.in/libgit2/git2go.v27 -p1 < patch/git2go.v27.patch >/dev/null; then \
        patch -d /home/ant/go/src/gopkg.in/libgit2/git2go.v27 -p1 < patch/git2go.v27.patch; \
fi
make -C /home/ant/go/src/gopkg.in/libgit2/git2go.v27 install-static
make[1]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27'
./script/build-libgit2-static.sh
+ VENDORED_PATH=vendor/libgit2
+ cd vendor/libgit2
+ mkdir -p install/lib
+ mkdir -p build
+ cd build
+ cmake -DTHREADSAFE=ON -DBUILD_CLAR=OFF -DBUILD_SHARED_LIBS=OFF -DCMAKE_C_FLAGS=-fPIC -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=../install -DUSE_EXT_HTTP_PARSER=OFF -DUSE_HTTPS=OFF -DUSE_NSEC=OFF -DUSE_SSH=OFF -DCURL=OFF -DUSE_GSSAPI=OFF -DUSE_BUNDLED_ZLIB=ON ..
  details.  Use the cmake_policy command to set the policy and suppress this
  Quoted variables like "HAVE_QSORT_R_BSD" will no longer be dereferenced
  when the policy is set to NEW.  Since the policy is not set the OLD
  behavior will be used.
Call Stack (most recent call first):
  src/CMakeLists.txt:60 (CHECK_PROTOTYPE_DEFINITION)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find HTTP_Parser (missing: HTTP_PARSER_INCLUDE_DIR HTTP_PARSER_LIBRARY) 
-- http-parser version 2 was not found or disabled; using bundled 3rd-party sources.
-- LIBSSH2 not found. Set CMAKE_PREFIX_PATH if it is installed outside of the default search path.
-- Enabled features:
 * threadsafe, threadsafe support
 * SHA, using CollisionDetection
 * http-parser, http-parser support (bundled)
 * zlib, using bundled zlib

-- Disabled features:
 * debugpool, debug pool allocator
 * tracing, tracing support
 * cURL, cURL for HTTP proxy support
 * HTTPS, no support
 * SSH, SSH transport support
 * SPNEGO, SPNEGO authentication support
 * iconv, iconv encoding conversion support

-- Configuring done
-- Generating done
-- Build files have been written to: /home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build
+ cmake --build .
make[2]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
make[3]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
make[4]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
make[4]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
[  1%] Built target http-parser
make[4]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
make[4]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
[  7%] Built target zlib
make[4]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
make[4]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
[100%] Built target git2internal
make[4]: Entering directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
make[4]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
[100%] Built target git2
make[3]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
make[2]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build'
go install --tags "static" ./...
# pkg-config --cflags --static -- /home/ant/go/src/gopkg.in/libgit2/git2go.v27/vendor/libgit2/build/libgit2.pc
pkg-config: exec: "pkg-config": executable file not found in $PATH
Makefile:14: recipe for target 'install-static' failed
make[1]: *** [install-static] Error 2
make[1]: Leaving directory '/home/ant/go/src/gopkg.in/libgit2/git2go.v27'
Makefile:34: recipe for target 'build-libgit2' failed
make: *** [build-libgit2] Error 2
```

--- a/README.md
+++ b/README.md
@@ -13,7 +13,7 @@ gitin is a minimalist tool that lets you explore a git repository from command l
 - Download latest release from [here](https://github.com/isacikgoz/gitin/releases)
 - Or, manually download it with `go get -d github.com/isacikgoz/gitin`
 - `cd` into `$GOPATH/src/github.com/isacikgoz/gitin`
-- build with `make install`
+- build with `make install` (`cmake` and `pkg-config` are required)
 
 ### Mac using brew 
 ```
@@ -51,7 +51,7 @@ Commands:
 ## Development Requirements
 - Requires gitlib2 v27 and `git2go`. See the project homepages for build instructions.
   1. download git2go; `go get -d gopkg.in/libgit2/git2go.v27`
-  2. make sure you have `cmake` `libssl-dev` installed
+  2. make sure you have `cmake`, `pkg-confid` and `libssl-dev` installed
   3. `cd` into `$GOPATH/src/gopkg.in/libgit2/git2go.v27`
   4. initialize submodules by running `git submodule update --init`
   5. change the libigt2 version to your version (in this case its 0.27) in the install script (`script/install-libgit2.sh`)

--- a/README.md
+++ b/README.md
@@ -17,7 +17,7 @@ gitin is a minimalist tool that lets you explore a git repository from command l
 
 ### Mac using brew 
 ```
-brew tap jeroenknoops/tap
+brew tap isacikgoz/gitin
 brew install gitin
 ```
 

--- a/cli/branch.go
+++ b/cli/branch.go
@@ -83,6 +83,7 @@ func branchPrompt(r *git.Repository, opts *PromptOptions) error {
 			return nil
 		}
 		cmd := exec.Command("git", "branch", "-d", b.Name)
+		cmd.Dir = r.AbsPath
 		if err := cmd.Run(); err == nil {
 			log.Error(err)
 		}
@@ -108,6 +109,7 @@ func branchPrompt(r *git.Repository, opts *PromptOptions) error {
 	if err == nil {
 		screenbuf.Clear(os.Stdin)
 		cmd := exec.Command("git", "checkout", r.Branches[i].Name)
+		cmd.Dir = r.AbsPath
 		cmd.Stdout = os.Stdout
 		cmd.Stderr = os.Stderr
 		cmd.Stdin = os.Stdin

--- a/cli/commit.go
+++ b/cli/commit.go
@@ -3,10 +3,7 @@ package cli
 import (
 	"os"
 	"os/exec"
-	"regexp"
-	"strings"
 
-	"github.com/fatih/color"
 	"github.com/isacikgoz/gitin/git"
 )
 
@@ -23,31 +20,21 @@ func CommitBuilder(r *git.Repository, opts *CommitOptions) error {
 	}
 	return commitPrompt(r, opts)
 }
+
 func commitPrompt(r *git.Repository, opts *CommitOptions) error {
-	cmd := exec.Command("git", "commit", "--edit", "--quiet")
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	cmd.Stdin = os.Stdin
-	ok := false
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-	if err := cmd.Wait(); err == nil {
-		ok = true
-	}
-	if err := r.InitializeStatus(); err != nil {
-		return err
-	}
-	if ok {
-		if err := popCommitStat(r.LastCommitHash()); err != nil {
-			return err
-		}
-	}
-	return NoErrRecurse
+	args := []string{"--edit", "--quiet"}
+	return execCommit(r, args)
 }
 
 func commitAmend(r *git.Repository) error {
-	cmd := exec.Command("git", "commit", "--amend", "--quiet")
+	args := []string{"--amend", "--quiet"}
+	return execCommit(r, args)
+}
+
+func execCommit(r *git.Repository, args []string) error {
+	args = append([]string{"commit"}, args...)
+	cmd := exec.Command("git", args...)
+	cmd.Dir = r.AbsPath
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	cmd.Stdin = os.Stdin
@@ -62,89 +49,9 @@ func commitAmend(r *git.Repository) error {
 		return err
 	}
 	if ok {
-		if err := popCommitStat(r.LastCommitHash()); err != nil {
+		if err := popGitCmd(r, r.LastCommitArgs()); err != nil {
 			return err
 		}
 	}
 	return NoErrRecurse
-}
-
-func popCommitStat(hash string) error {
-	os.Setenv("LESS", "-RC")
-
-	cmd := exec.Command("git", "show", "--stat", hash)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	cmd.Stdin = os.Stdin
-
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-	if err := cmd.Wait(); err != nil {
-		return err
-	}
-	return nil
-}
-
-func popCommitDiff(hash string) error {
-	os.Setenv("LESS", "-RCS")
-
-	cmd := exec.Command("git", "diff", hash)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	cmd.Stdin = os.Stdin
-
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-	if err := cmd.Wait(); err != nil {
-		return err
-	}
-	return nil
-}
-
-func colorizeStat(input string) string {
-	var output string
-	yellow := color.New(color.FgYellow)
-	re := regexp.MustCompile(`\s\|\s+[\d|Bin|bin]+\s+`)
-	rc := regexp.MustCompile(`commit\s+\w{40}`)
-	rl := regexp.MustCompile(`\r?\n`)
-	rs := regexp.MustCompile(`\d+\s+.+\s+\w+$`)
-	lines := rl.Split(input, -1)
-	if len(lines) > 0 && rc.MatchString(lines[0]) {
-		labelandhash := rc.FindString(lines[0])
-		output = output + yellow.Sprint(labelandhash) + "\n"
-	}
-	for _, line := range lines[1:] {
-		if re.MatchString(line) {
-			a := re.Split(line, -1)
-			file := a[0]
-			stat := a[1]
-			separator := " | "
-			concatStat := strings.Replace(line, a[1], "", 1)
-			concatSeparator := strings.Replace(concatStat, separator, "", 1)
-			change := strings.Replace(concatSeparator, a[0], "", 1)
-
-			line = file + yellow.Sprint(separator) + change + paintStats(stat, rs)
-
-		}
-		output = output + "\n" + line
-	}
-	return output
-}
-
-func paintStats(input string, re *regexp.Regexp) string {
-	var output string
-	green := color.New(color.FgGreen)
-	red := color.New(color.FgRed)
-	if re.MatchString(input) {
-		output = input
-	} else {
-		pluses := strings.Count(input, "+")
-		minuses := strings.Count(input, "-")
-		output = strings.Repeat(green.Sprint("+"), pluses) +
-			strings.Repeat(red.Sprint("-"), minuses)
-	}
-
-	return output
 }

--- a/cli/log.go
+++ b/cli/log.go
@@ -95,7 +95,7 @@ func logPrompt(r *git.Repository, opts *PromptOptions, commits []*git.Commit) er
 	}
 	kset['s'] = func(in interface{}, chb chan bool, index int) error {
 		screenbuf.Clear(os.Stdin)
-		if err := popCommitStat(commits[index].Hash); err != nil {
+		if err := popGitCmd(r, []string{"show", "--stat", commits[index].Hash}); err != nil {
 			return err
 		}
 		chb <- true
@@ -109,7 +109,7 @@ func logPrompt(r *git.Repository, opts *PromptOptions, commits []*git.Commit) er
 	}
 	kset['d'] = func(in interface{}, chb chan bool, index int) error {
 		screenbuf.Clear(os.Stdin)
-		if err := popCommitDiff(commits[index].Hash); err != nil {
+		if err := popGitCmd(r, []string{"diff", commits[index].Hash}); err != nil {
 			return err
 		}
 		chb <- true

--- a/cli/prompt.go
+++ b/cli/prompt.go
@@ -2,11 +2,10 @@ package cli
 
 import (
 	"errors"
-	"io"
 	"os"
 	"os/exec"
 
-	log "github.com/sirupsen/logrus"
+	"github.com/isacikgoz/gitin/git"
 )
 
 // PromptOptions is the common options for building a prompt
@@ -22,22 +21,15 @@ var (
 	NoErrRecurse error = errors.New("catch")
 )
 
-func popMore(in string) error {
+func popGitCmd(r *git.Repository, args []string) error {
 	os.Setenv("LESS", "-RCS")
-	cmd := exec.Command("less")
-	stdin, err := cmd.StdinPipe()
-	if err != nil {
-		log.Fatal(err)
-	}
+	cmd := exec.Command("git", args...)
+	cmd.Dir = r.AbsPath
+
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	defer func() {
-		cmd.Stdin = os.Stdin
-	}()
-	go func() {
-		defer stdin.Close()
-		io.WriteString(stdin, in)
-	}()
+	cmd.Stdin = os.Stdin
+
 	if err := cmd.Start(); err != nil {
 		return err
 	}

--- a/cli/stat.go
+++ b/cli/stat.go
@@ -1,15 +1,11 @@
 package cli
 
 import (
-	"io"
 	"os"
-	"os/exec"
 
 	"github.com/isacikgoz/gitin/git"
 	"github.com/isacikgoz/promptui"
 	"github.com/isacikgoz/promptui/screenbuf"
-
-	log "github.com/sirupsen/logrus"
 )
 
 func statPrompt(r *git.Repository, c *git.Commit, opts *PromptOptions) error {
@@ -18,7 +14,7 @@ func statPrompt(r *git.Repository, c *git.Commit, opts *PromptOptions) error {
 	if err != nil {
 		return err
 	}
-
+	deltas := diff.Deltas()
 	kset := make(map[rune]promptui.CustomFunc)
 	kset['q'] = func(in interface{}, chb chan bool, index int) error {
 		screenbuf.Clear(os.Stdin)
@@ -29,7 +25,7 @@ func statPrompt(r *git.Repository, c *git.Commit, opts *PromptOptions) error {
 
 	prompt := promptui.Select{
 		Label:       c,
-		Items:       diff.Deltas(),
+		Items:       deltas,
 		HideHelp:    opts.HideHelp,
 		Size:        opts.Size,
 		Templates:   statTemplate(c),
@@ -46,34 +42,9 @@ func statPrompt(r *git.Repository, c *git.Commit, opts *PromptOptions) error {
 			Size:     opts.Size,
 			HideHelp: opts.HideHelp,
 		}
-		if err = popLess(r, c, diff.Deltas()[i].PatchString()); err == nil {
+		if err = popGitCmd(r, deltas[i].FileStatArgs(c)); err == NoErrRecurse {
 			return statPrompt(r, c, o)
 		}
-	}
-	return nil
-}
-
-func popLess(r *git.Repository, c *git.Commit, in string) error {
-	os.Setenv("LESS", "-RCS")
-	cmd := exec.Command("less")
-	stdin, err := cmd.StdinPipe()
-	if err != nil {
-		log.Fatal(err)
-	}
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	defer func() {
-		cmd.Stdin = os.Stdin
-	}()
-	go func() {
-		defer stdin.Close()
-		io.WriteString(stdin, in)
-	}()
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-	if err := cmd.Wait(); err != nil {
-		// return statPrompt(r, c, opts)
 	}
 	return nil
 }

--- a/cli/status.go
+++ b/cli/status.go
@@ -118,7 +118,7 @@ func statusPrompt(r *git.Repository, opts *PromptOptions) error {
 			Size:     opts.Size,
 			HideHelp: opts.HideHelp,
 		}
-		if err := popMore(r.Status.Entries[i].Patch()); err == NoErrRecurse {
+		if err := popGitCmd(r, r.Status.Entries[i].FileStatArgs()); err == NoErrRecurse {
 			return statusPrompt(r, o)
 		}
 	}

--- a/git/commit.go
+++ b/git/commit.go
@@ -9,6 +9,7 @@ import (
 	"time"
 
 	"github.com/justincampbell/timeago"
+	log "github.com/sirupsen/logrus"
 	lib "gopkg.in/libgit2/git2go.v27"
 )
 
@@ -134,6 +135,7 @@ func (r *Repository) failOverShallow(opts *CommitLoadOptions) ([]*Commit, error)
 	}
 	defer file.Close()
 	cmd := exec.Command("git", "rev-list", "--all")
+	cmd.Dir = r.AbsPath
 	out, err := cmd.Output()
 	if err != nil {
 		return nil, err
@@ -213,7 +215,6 @@ func (r *Repository) Diff(c *Commit) (*Diff, error) {
 		return nil, err
 	}
 	defer cTree.Free()
-
 	var pTree *lib.Tree
 	if c.commit.ParentCount() > 0 {
 		if pTree, err = c.commit.Parent(0).Tree(); err != nil {
@@ -311,6 +312,7 @@ func (r *Repository) revlist(from, to *lib.Oid) ([]*Commit, error) {
 	commits := make([]*Commit, 0)
 
 	cmd := exec.Command("git", "rev-list", from.String()+".."+to.String())
+	cmd.Dir = r.AbsPath
 	out, err := cmd.Output()
 	if err != nil {
 		return commits, err
@@ -424,6 +426,7 @@ func (r *Repository) LastCommitStat() string {
 	}
 	hash := commit.AsObject().Id().String()
 	cmd := exec.Command("git", "show", "--stat", hash)
+	cmd.Dir = r.AbsPath
 	out, err := cmd.Output()
 	if err != nil {
 		return err.Error()
@@ -431,11 +434,14 @@ func (r *Repository) LastCommitStat() string {
 	return string(out)
 }
 
-// LastCommitHash get the HEAD's target hash
-func (r *Repository) LastCommitHash() string {
+// LastCommitArgs returns the args for show stat
+func (r *Repository) LastCommitArgs() []string {
 	head, err := r.repo.Head()
 	if err != nil {
-		return "error reading last commit"
+		log.Error(err)
+		return nil
 	}
-	return string(head.Target().String())
+	hash := string(head.Target().String())
+	args := []string{"show", "--stat", hash}
+	return args
 }

--- a/git/diff.go
+++ b/git/diff.go
@@ -45,6 +45,21 @@ func (d *DiffDelta) String() string {
 	return s
 }
 
+// FileStatArgs returns git command args for getting diff
+func (d *DiffDelta) FileStatArgs(c *Commit) []string {
+	args := []string{"diff"}
+	if c.commit.Parent(0) == nil {
+		args = []string{"show", "--oneline", "--patch"}
+		args = append(args, c.Hash)
+	} else {
+		parent := c.commit.Parent(0).AsObject().Id().String()
+		args = append(args, parent+".."+c.Hash)
+	}
+	args = append(args, d.OldFile.Path)
+
+	return args
+}
+
 // colorize the plain diff text collected from system output
 // the style is near to original diff command
 func colorizeDiff(original string) (colorized []string) {

--- a/git/repository.go
+++ b/git/repository.go
@@ -1,6 +1,9 @@
 package git
 
 import (
+	"errors"
+	"path/filepath"
+
 	lib "gopkg.in/libgit2/git2go.v27"
 )
 
@@ -29,14 +32,14 @@ type Remote struct {
 
 // Open the repository from given path and return Repository pointer
 func Open(path string) (*Repository, error) {
-	r, err := lib.OpenRepository(path)
+	r, absPath, err := initRepoFromPath(path)
 	if err != nil {
 		return nil, err
 	}
 	repo := &Repository{
 		RepoID:  "",
 		Name:    "",
-		AbsPath: path,
+		AbsPath: absPath,
 		repo:    r,
 	}
 	if err := repo.loadStatus(); err != nil {
@@ -86,4 +89,19 @@ func (r *Repository) InitializeCommits(opts *CommitLoadOptions) error {
 	}
 	r.Commits = commits
 	return nil
+}
+
+func initRepoFromPath(path string) (*lib.Repository, string, error) {
+	walk := path
+	for {
+		r, err := lib.OpenRepository(walk)
+		if err == nil {
+			return r, walk, err
+		}
+		walk = filepath.Dir(walk)
+		if walk == "/" {
+			break
+		}
+	}
+	return nil, walk, errors.New("cannot load a git repository from " + path)
 }

--- a/git/status.go
+++ b/git/status.go
@@ -138,18 +138,30 @@ func (e *StatusEntry) String() string {
 }
 
 // Patch return the diff of the entry
-func (e *StatusEntry) Patch() string {
+func (r *Repository) Patch(e *StatusEntry) string {
 	var cmd *exec.Cmd
 	if e.statusEntryType == StatusEntryTypeUntracked {
 		cmd = exec.Command("git", "diff", "--no-index", "/dev/null", e.diffDelta.NewFile.Path)
 	} else {
 		cmd = exec.Command("git", "diff", e.diffDelta.OldFile.Path)
 	}
+	cmd.Dir = r.AbsPath
 	out, err := cmd.CombinedOutput()
 	if err != nil {
 		log.Warn(err.Error())
 	}
 	return strings.Join(colorizeDiff(string(out)), "\n")
+}
+
+// FileStatArgs returns git command args for getting diff
+func (e *StatusEntry) FileStatArgs() []string {
+	var args []string
+	if e.statusEntryType == StatusEntryTypeUntracked {
+		args = []string{"diff", "--no-index", "/dev/null", e.diffDelta.NewFile.Path}
+	} else {
+		args = []string{"diff", e.diffDelta.OldFile.Path}
+	}
+	return args
 }
 
 // StatusEntryString returns entry status in pretty format
@@ -193,6 +205,7 @@ func (e *StatusEntry) Indexed() bool {
 // AddEntry is the wrapper of "git add /path/to/file" command
 func (r *Repository) AddEntry(e *StatusEntry) error {
 	cmd := exec.Command("git", "add", "--", e.diffDelta.OldFile.Path)
+	cmd.Dir = r.AbsPath
 	if err := cmd.Run(); err != nil {
 		return err
 	}
@@ -202,6 +215,7 @@ func (r *Repository) AddEntry(e *StatusEntry) error {
 // ResetEntry is the wrapper of "git reset path/to/file" command
 func (r *Repository) ResetEntry(e *StatusEntry) error {
 	cmd := exec.Command("git", "reset", "HEAD", "--", e.diffDelta.OldFile.Path)
+	cmd.Dir = r.AbsPath
 	if err := cmd.Run(); err != nil {
 		return err
 	}
@@ -211,6 +225,7 @@ func (r *Repository) ResetEntry(e *StatusEntry) error {
 // AddAll is the wrapper of "git add ." command
 func (r *Repository) AddAll() error {
 	cmd := exec.Command("git", "add", ".")
+	cmd.Dir = r.AbsPath
 	if err := cmd.Run(); err != nil {
 		return err
 	}
@@ -220,6 +235,7 @@ func (r *Repository) AddAll() error {
 // ResetAll is the wrapper of "git reset" command
 func (r *Repository) ResetAll() error {
 	cmd := exec.Command("git", "reset", "--mixed")
+	cmd.Dir = r.AbsPath
 	if err := cmd.Run(); err != nil {
 		return err
 	}

--- a/main.go
+++ b/main.go
@@ -36,7 +36,7 @@ var (
 
 func main() {
 
-	pin.Version("gitin version 0.1.1")
+	pin.Version("gitin version 0.1.2")
 	pin.CommandLine.HelpFlag.Short('h')
 	pin.CommandLine.VersionFlag.Short('v')
 	pin.Parse()

